packages/text-bitmap/src/utils/generateFillStyle.ts

```diff
@@ -0,0 +1,148 @@
+import { TextStyle, TextMetrics, TEXT_GRADIENT } from '@pixi/text';
+
+/**
+ * Generates the fill style. Can automatically generate a gradient based on the fill style being an array
+ *
+ * @private
+ * @param {object} style - The style.
+ * @param {string[]} lines - The lines of text.
+ * @return {string|number|CanvasGradient} The fill style
+ */
+export function generateFillStyle(
+    canvas: HTMLCanvasElement,
+    context: CanvasRenderingContext2D,
+    style: TextStyle,
+    resolution: number,
+    lines: string[],
+    metrics: TextMetrics
+): string|CanvasGradient|CanvasPattern
+{
```

Comment:
I'm worried that we have repeated code in generateFillStyle between here and 'Text'?

Reply:
Same here, let’s not balloon scope. We can find a better way to shared some of the internal rendering guys. Maybe adding TODO in the code would be a good reminder.

Reply:
I added a TODO comment 👍